lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -200,34 +209,66 @@ public SystemAccessControl create(Map<String, String> config)
-            Injector injector = bootstrap
+                    binder -> {
+                        ConfigurationFactory configurationFactory = new ConfigurationFactory(config);
+                        boolean isUsingRest = bindRules(binder,
+                                configurationFactory.build(FileBasedAccessControlConfig.class));
+                        configBinder(binder).bindConfig(FileBasedAccessControlConfig.class);
+                        if (isUsingRest) {
+                            configBinder(binder).bindConfig(HttpClientConfig.class, HTTP_CLIENT_NAME);
+                        }
+                    });
+            injector = bootstrap
                     .doNotInitializeLogging()
                     .setRequiredConfigurationProperties(config)
                     .initialize();
-            FileBasedAccessControlConfig fileBasedAccessControlConfig = injector.getInstance(FileBasedAccessControlConfig.class);
-            String configFileName = fileBasedAccessControlConfig.getConfigFile().getPath();
+            FileBasedAccessControlConfig fileBasedAccessControlConfig = injector
+                    .getInstance(FileBasedAccessControlConfig.class);
+            AccessControlRulesProvider<FileBasedSystemAccessControlRules> rulesProvider = injector
+                    .getInstance(Key.get(new TypeLiteral<AccessControlRulesProvider<FileBasedSystemAccessControlRules>>() {}));
+            String configFilePath = fileBasedAccessControlConfig.getConfigFilePath();
 
             if (config.containsKey(SECURITY_REFRESH_PERIOD)) {
                 Duration refreshPeriod;
                 try {
                     refreshPeriod = fileBasedAccessControlConfig.getRefreshPeriod();
                 }
                 catch (IllegalArgumentException e) {
-                    throw invalidRefreshPeriodException(config, configFileName);
+                    throw invalidRefreshPeriodException(config, configFilePath);
                 }
                 if (refreshPeriod.toMillis() == 0) {
-                    throw invalidRefreshPeriodException(config, configFileName);
+                    throw invalidRefreshPeriodException(config, configFilePath);
                 }
                 return ForwardingSystemAccessControl.of(memoizeWithExpiration(
                         () -> {
-                            log.info("Refreshing system access control from %s", configFileName);
-                            return create(configFileName);
+                            log.info("Refreshing system access control from %s", configFilePath);
+                            return create(rulesProvider);
                         },
                         refreshPeriod.toMillis(),
                         MILLISECONDS));
             }
-            return create(configFileName);
+            return create(rulesProvider);
+        }
+
+        private boolean bindRules(Binder binder, FileBasedAccessControlConfig configuration)
```

Comment:
I don't like `bindRules` returns a `boolean`. It should only bind things. Maybe this method should be inlined?